test/unit/spec/default_spec.rb

```diff
@@ -0,0 +1,305 @@
+    '2012R2': {
+      splunk: 'https://download.splunk.com/products/splunk/releases/6.3.4/windows/splunk-6.3.4-cae2458f4aef-x64-release.msi',
+      universal_forwarder: 'https://download.splunk.com/products/universalforwarder/releases/6.3.4/windows/splunkforwarder-6.3.4-cae2458f4aef-x64-release.msi'
+    }
+  },
+  suse: { # Generic Linux
+    '12.0': {
+      splunk: 'https://download.splunk.com/products/splunk/releases/6.3.4/linux/splunk-6.3.4-cae2458f4aef-Linux-x86_64.tgz',
+      universal_forwarder: 'https://download.splunk.com/products/universalforwarder/releases/6.3.4/linux/splunkforwarder-6.3.4-cae2458f4aef-Linux-x86_64.tgz'
+    }
+  }
+}
+
+describe 'splunk_install' do
+  let(:test_params) { {} }
+  let(:platform) { 'redhat' }
+  let(:version) { '7.1' }
+  let(:do_before) {}
+  let(:run_state) { {} }
+  let(:chef_runner) do
+    parameters = { step_into: ['splunk_install'], platform: platform, version: version }
+    ChefSpec::SoloRunner.new(parameters) do |node|
+      node.set['test_parameters'] = test_params
+      node.set['run_state'].merge!(run_state)
+    end
+  end
+  let(:chef_run) do
+    do_before
+    chef_runner.converge('cerner_splunk_ingredient_test::unit_test')
+  end
+  subject { chef_run }
+
+  describe 'action :install' do
+    platform_urls.each do |platform, versions|
+      versions.each do |version, urls|
+        context "on #{platform} #{version}" do
+          let(:platform) { platform.to_s }
+          let(:version) { version.to_s }
+          context 'with the Splunk package' do
+            let(:test_params) { { build: 'cae2458f4aef', version: '6.3.4', accept_license: true } }
+            let(:package_dir) { "./test/unit/.cache/#{filename_from_url(urls[:splunk])}" }
+
+            it 'should set the run state' do
+              expect(chef_run.node.run_state['splunk_ingredient']['current_installation']).to eq(
+                name: 'splunk',
+                package: :splunk,
+                version: '6.3.4',
+                build: 'cae2458f4aef',
+                x64: true
+              )
+            end
+
+            it { is_expected.to create_remote_file(package_dir).with(source: urls[:splunk]) }
+
+            case platform
+            when :redhat then it { is_expected.to install_rpm_package('splunk').with(source: package_dir) }
+            when :ubuntu then it { is_expected.to install_dpkg_package('splunk').with(source: package_dir) }
+            when :windows
+              context 'accepting the license agreeemnt' do
+                it do
+                  is_expected.to install_windows_package('Splunk Enterprise').with(
+                    options: 'LAUNCHSPLUNK=0 INSTALL_SHORTCUT=0 AGREETOLICENSE=Yes'
+                  )
+                end
+              end
+            else it { is_expected.to extract_local_tar_extract(package_dir).with(target_dir: '/opt/splunk') }
+            end
+
+            context 'when Splunk is already installed' do
+              let(:do_before) do
```

Comment:
A before block would run before each test, but oddly not before the subject (the chef run). I need to be able to stub things before the chef run too, so this let is called before the chef run is converged.